backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/objcexport/ObjCExportHeaderGenerator.kt

```diff
@@ -973,3 +1018,50 @@ private fun ObjCExportNamer.ClassOrProtocolName.toNameAttributes(): List<String>
 
 private fun swiftNameAttribute(swiftName: String) = "swift_name(\"$swiftName\")"
 private fun objcRuntimeNameAttribute(name: String) = "objc_runtime_name(\"$name\")"
+
+interface TypeParameterProvider{
+    fun typeAvailable(kotlinType: KotlinType):Boolean
+}
+
+internal class ClassParameterProvider(container:DeclarationDescriptor): TypeParameterProvider {
+    constructor(method: FunctionDescriptor) : this(method.containingDeclaration)
+    private val nameSet:Set<String>
+    init {
+        nameSet = mutableSetOf()
+        if(container is ClassDescriptor && !container.isInterface) {
+            container.declaredTypeParameters.forEach {
```

Comment:
Does this work for inner class when outer class has its own type parameters?
If yes, does this work when there is a name clash between inner and outer type parameters?

Reply:
It did not, but updated. The outer class type params aren't visible to the inner class, so they're all 'id' (Any? to swift), but they do work.